src/lgdo/lh5/core.py

```diff
@@ -97,6 +98,10 @@ def read(
         Decompress data encoded with LGDO's compression routines right
         after reading. The option has no effect on data encoded with HDF5
         built-in filters, which is always decompressed upstream by HDF5.
+    use_metadata
```

Comment:
a) I think during processing where you take e.g. only one dataset like the waveforms, the processing chain might want to avoid loading the metadata since it is a little slower (though negligibly in the grand scheme of things). Some people in the collaboration like having additional knobs.

b) The way I wrote the metadata is rather simple in that it loops over the whole HDF5 file structure every time when you write the metadata. This was opposed to the more complicated thing where it adds and subtracts fields from a dict in memory before writing and keeps track separately basically. I did this to avoid complications where you could write to the file not using LH5Store or using a different instance or other complicated things. I was very unsure whether I could guarantee that the metadata that is written matches the actual file structure which is what is needed for loading data and so chose to do the simple and robust thing. 

This meant that if I re-write the metadata after every write to the file, then the metadata writing will take an extremely long time compared to the size of a dataset that I might add to the file. So I decided instead to let the user decide when to write the metadata to the file, which they should do after they have finished writing all the data they want to the file. It's possible that this is also fragile - if the user writes something to a file, then tries to load it from the file - it won't be found in the metadata of the file - but I think that I wrote it to use the file attributes as a backup, but probably worth me adding as a unit test.

Reply:
Actually I might have thought of this case and fixed it here: https://github.com/legend-exp/legend-pydataobj/pull/94/files#diff-560aa62aa8588dc1d5273e460904246108340a4a548df758010365630918106dR366-R374 Not 100% sure though, can do some testing.